onnxruntime/core/providers/snpe/snpe_lib.cc

```diff
@@ -0,0 +1,467 @@
+
+  user_buffer_attr_table_.emplace(name, UserBufferAttribute(bufSize, strides, user_buffer_encoding));
+
+  return true;
+}
+
+bool SnpeLib::SetupUserBufferAttributes(const std::vector<std::string>& tensor_names) {
+  for (size_t i = 0; i < tensor_names.size(); ++i) {
+    bool rt = SetupUserBufferAttribute(tensor_names.at(i));
+    if (!rt) {
+      LOGS_DEFAULT(ERROR) << "Failed to set user buffer attribute for: " << tensor_names.at(i);
+      return false;
+    }
+  }
+
+  return true;
+}
+
+bool SnpeLib::CheckInputsSize(const std::vector<std::string>& input_tensor_names,
+                              const std::vector<int64_t>& input_sizes) {
+  size_t elementSize = 1;
+  for (size_t i = 0; i < input_tensor_names.size(); ++i) {
+    auto input_shape = snpe_->getInputDimensions(input_tensor_names.at(i).c_str());
+    if (!input_shape) {
+      LOGS_DEFAULT(ERROR) << "Snpe cannot get input shape for input name: " << input_tensor_names.at(i);
+      return false;
+    }
+    int64_t input_size = CalcSizeFromDims((*input_shape).getDimensions(), (*input_shape).rank(), elementSize);
+    if (input_sizes.at(i) != input_size) {
+      LOGS_DEFAULT(ERROR) << "Input size mismatch for : " << input_tensor_names.at(i)
+                          << " size on dlc: " << input_size
+                          << " size on onnx: " << input_sizes.at(i);
+      return false;
+    }
+  }
+  return true;
+}
+
+bool SnpeLib::SetupInputTensors(const std::vector<std::string>& input_tensor_names) {
+  input_tensor_map_.clear();
+  input_tensors_.clear();
+  input_tensors_.resize(input_tensor_names.size());
+  for (size_t i = 0; i < input_tensor_names.size(); ++i) {
+    auto input_shape = snpe_->getInputDimensions(input_tensor_names.at(i).c_str());
+    if (!input_shape) {
+      LOGS_DEFAULT(ERROR) << "Snpe cannot get input shape for input name: " << input_tensor_names.at(i);
+      input_tensor_map_.clear();
+      input_tensors_.clear();
+      return false;
+    }
+    input_tensors_[i] = zdl::SNPE::SNPEFactory::getTensorFactory().createTensor(*input_shape);
+    zdl::DlSystem::ITensor* input_tensor = input_tensors_[i].get();
+    if (!input_tensor) {
+      LOGS_DEFAULT(ERROR) << "Snpe cannot create ITensor!";
+      input_tensor_map_.clear();
+      input_tensors_.clear();
+      return false;
+    }
+    input_tensor_map_.add(input_tensor_names.at(i).c_str(), input_tensor);
+  }
+  return true;
+}
+
+bool SnpeLib::InitializeSnpe(zdl::DlContainer::IDlContainer* container,
+                             const std::vector<std::string>& output_tensor_names,
+                             const std::vector<std::string>& input_tensor_names,
+                             const std::vector<int64_t>& input_sizes,
+                             const SnpeRuntimeOptions& snpe_settings) {
+  if (!snpe_settings.GetRuntimeTarget().IsAvailable()) {
+    LOGS_DEFAULT(INFO) << "Provided runtime target is not available" << snpe_settings.GetRuntimeTarget().ToString();
```

Comment:
should these functions return Status instead of bool? then this extra information can be included in the return value.

Reply:
thanks for updating. looks like there are still some other functions that return bool to indicate success